src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +32,11 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToDouble(i -> numbers.get(i) - i % 2)
+                .filter(num -> num % 2 != 0)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException());
```

Comment:
if you read again "try to avoid common mistakes" you will find this 
```suggestion
                .getAsDouble();
```